### PR TITLE
bug(query):  single matching regex queries get split into 2 clusters

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 

For Queries like `histogram_quantile(0.50, sum(rate(namespace_lookup_time_ns{_ws_="aci-telemetry",_ns_=~"query-server.mosaic-pie-filodb-query-service.mosaic-dev2"}[5m])) by (le, _ns_))` ,`sum(rate(namespace_lookup_time_ns{_ws_="aci-telemetry",_ns_=~"query-server.mosaic-pie-filodb-query-service.mosaic-dev2"}[5m]))` gets dispatched to remote cluster and histogram_quantile is applied locally

**New behavior :**
Since there is only one matching regex, dispatch the entire query to the remote cluster

### CODE REVIEW

**Review by tjackpaul — APPROVED** [review]
lgtm.. thanks for the quick pr @TanviBhavsar .. one minor comment is to revisit `LogicalPlan.getNonMetricShardKeyFilters` usage in multiple flow of the code

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala [hunk 1]
```diff
@@ -33,6 +33,17 @@ class ShardKeyRegexPlanner(dataset: Dataset,
...
   override val schemas = Schemas(dataset.schema)
 
+  /**
+   * Returns true when regex has single matching value
+   * Example: sum(test1{_ws_ = "demo", _ns_ =~ "App-1"}) + sum(test2{_ws_ = "demo", _ns_ =~ "App-1"})
+   */
+  def hasSingleShardKeyMatch(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]]) = {
+    val shardKeyMatchers = nonMetricShardKeyFilters.map(shardKeyMatcher(_))
+    shardKeyMatchers.forall(_.size == 1) &&
+      shardKeyMatchers.forall(_.head.toSet.sameElements(shardKeyMatchers.head.head.toSet))
```
Comment (tjackpaul) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:43]: @vishramachandran any concerns over this set conversion and comparison?

**Review by tjackpaul — APPROVED** [review]

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala (hunk 1 above)
Comment (vishramachandran) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala:43]: I have concerns with `head.head` - susceptible to exception. Let us discuss.

**Comment by TanviBhavsar** [comment]
> lgtm.. thanks for the quick pr @TanviBhavsar .. one minor comment is to revisit `LogicalPlan.getNonMetricShardKeyFilters` usage in multiple flow of the code

As discussed we need it for Binary Join and nested queries

**Pull request merged**